

```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(qiimer)
library(vegan)
library(ape)
library(usedist)

## Visualization packages
library(pander)
library(kableExtra)

library(pheatbuilder)
library(ggbeeswarm)
library(ggsci)
library(viridis)
library(wesanderson)
library(RColorBrewer)

# stats packages
library(adonisplus)
library(nlme)
library(emmeans) # for lmer post-hoc tests
library(broom)

library(pwr) # for power analysis
```

```{r}
se <- function(x) sd(x)/sqrt(length(x))

logit <- function(x) { log(x / (1-x)) }

tidy_lmer <- function(lmer_test) {
  mod <- summary(lmer_test)
  data.frame(term  = rownames(mod$tTable), mod$tTable, row.names=NULL)
}

tidy_lmer2 <- function(lmer_test, term_string) {
  form1 <- as.formula(paste("pairwise ~", study_group))
  mod <- anova(lm_test)
  bind_rows(data.frame(contrast = rownames(mod), mod, row.names=NULL) %>%
              rename(p.value="Pr..F.", df="Df"),
            data.frame(emmeans(lm_test, form1, adjust="tukey")$contrasts)
            ) %>%
    #mutate(AIC = AIC(lm_test)) %>%
    mutate(estimate = estimate * -1) %>%
    select(contrast, p.value, everything())
}

# tidy_lm_posthoc <- function(lm_test, study_group) {
#   form1 <- as.formula(paste("pairwise ~", study_group))
#   mod <- anova(lm_test)
#   bind_rows(data.frame(contrast = rownames(mod), mod, row.names=NULL) %>%
#               rename(p.value="Pr..F.", df="Df"),
#             data.frame(emmeans(lm_test, form1, adjust="tukey")$contrasts)
#             ) %>%
#     #mutate(AIC = AIC(lm_test)) %>%
#     mutate(estimate = estimate * -1) %>%
#     select(contrast, p.value, everything())
# }


tidy_lm_posthoc <- function(lmer_test, term_string) {
  form1 <- as.formula(paste("pairwise ~", term_string))
  mod <- anova(lmer_test)
  
  if(class(lmer_test) == "lm") {
    main_tidy <- tidy(mod)
  } else {
    main_tidy <- data.frame(term = rownames(mod), mod, row.names=NULL) %>% 
      rename(df=numDF)
  }
  
  bind_rows(main_tidy,
            data.frame(emmeans(lmer_test, form1, adjust="tukey")$contrasts) %>% rename(term = contrast, std.error=SE)
            ) %>%
    mutate(estimate = estimate * -1) %>%
    select(term, df, estimate, std.error,  p.value) ## can also add statistic and t.ratio columns if needed
}


theme_clean_pcoa <- function(){ 
    theme_bw() %+replace%    #replace elements we want to change
    theme(
      axis.text=element_blank(),
      axis.ticks=element_blank(),
      panel.grid = element_blank(), 
      aspect.ratio = 1,
      strip.background = element_blank()
    )
}
theme_clean <- function(){ 
    theme_bw() %+replace%    #replace elements we want to change
    theme(
      panel.grid = element_blank(), 
      strip.background = element_blank()
    )
}

kable_style <- function(data, col_name=p.value, threshold=0.05) {
  
  row_num <- nrow(data)
  
  ##substitute underscore with escaped underscores and remove na in p.value columns
  data_return <- data %>%
    select_all(~gsub("_", "\\\\_", .)) %>% ##need to escape the escape
    select_all(~gsub("#", "\\\\#", .)) %>% ##need to escape the escape
    mutate_if(function(x) is.character(x) | is.factor(x), ~gsub("_", " ", .)) %>%
    mutate_if(function(x) is.character(x) | is.factor(x), ~gsub("%", "\\\\%", .))
  
  ## highlight significant values
  col_name <- enquo(col_name)
  data_return <- data_return %>%
    mutate(!!col_name := cell_spec(signif(!!col_name, 2), "latex", bold = !!col_name<threshold))
  
  ##if Taxa is a column in the dataframe
  if(sum(grepl("Taxa", colnames(data_return))) > 0)  {
    data_return <- data_return %>%
      mutate(Taxa = gsub("[pcofgs]__", "", Taxa))
  }
  
  # ... should be column number
  if (row_num > 40) {
    data_return <- data_return %>%
      kable("latex", longtable = T, digits=2, booktabs=T, escape=F) %>%
      kable_styling(latex_options = c("repeat_header", "HOLD_position"), font_size = 7) %>%
      row_spec(0, bold = T, color="#7C0A02") #%>%
      #collapse_rows(columns = 1, valign = "top") 
    
  }
  else {
    data_return <- data_return %>%
      kable("latex", longtable = F, digits=2, booktabs=T, escape=F) %>%
      kable_styling(latex_options = c("scale_down", "repeat_header", "HOLD_position")) %>%
      row_spec(0, bold = T, color="#7C0A02")
    
    if(row_num > 1) { ##always collapse row unless there is only 1 row
      data_return <- data_return %>%
        collapse_rows(columns = 1, valign = "top")
    }
  }
  
  return(data_return)
  
}

```


```{r}
### =====================
###   define constants
### =====================

## This is assuming that you have created the 
data_dir <- "/Users/tanesc/Documents/betrnet/betrnet_data"

### minimum QC read count threshold
min_reads <- 1000

### rarefying subsample size 
richness_subsample_size <- 1000

## The number of permutations to do for PERMANOVA. You can start with 99 permutations to run faster while developing the code, then change it to 999 permutations for higher resolution.
perm <- 99 

### mapping file path
mapping_file_fp <- file.path(data_dir, "merged_Betrnet_metadata.tsv")

### demux counts file path
#demux_count_fp <- file.path(data_dir, "QIIME_output", "demux_stat", "per-sample-fastq-counts.csv")

### otu table file path
feature_table_fp <- file.path(data_dir, "denoising_results", "feature-table.tsv")

### taxonomic assignment file path
taxo_assignment_fp <- file.path(data_dir, "denoising_results", "taxonomy.tsv")

### unweighted UniFrac file path
uu_fp <- file.path(data_dir, "core_metrics_results", "uu.tsv")

### weighted UniFrac file path
wu_fp <- file.path(data_dir, "core_metrics_results", "wu.tsv")

### Faith phylogenetic diversity
faith_fp <- file.path(data_dir, "core_metrics_results", "faith_pd.tsv")
```


```{r}
## load the clinical information
clinical_data <- readxl::read_excel(file.path(data_dir, "BETRNET2_FINAL", "BETRNET2 LABELS.xlsx"), na=c(" ", "#NULL!")) %>%
  mutate(ANTIBIO = factor(ANTIBIO)) %>%
  mutate(ANTIBIO = fct_collapse(ANTIBIO, No = c("No", "Unknown"))) %>%
  select(STUDYID, ANTIBIO, FINAL_PATH, FINAL_HX, FINAL_WITHDRAW, AGE, SEX, BMI_RANGE, ASPIRIN, ETHNICITY, RACE, RACEOTHER, SMOKEEVER, FAMILY_HX, BELENGTHC, BELENGTHM, EPINCH, EFOLD, STATINS, HEARTBURN, ENDO_C_VALUE, ENDO_M_VALUE) %>% # Final Pathology Classification and Final Historical Pathology Classification
  rename(subject_id = STUDYID) %>%
  mutate(HEARTBURN = ifelse(HEARTBURN=="Unknown", NA, HEARTBURN)) %>%
  mutate(FINAL_HX = sub("N/A \\(Control\\)", "Control", FINAL_HX))
```


```{r}
### read mapping file

sample_types <- read_csv("../betrnet_data/sample_types.csv")
sample_type_order <- sample_types$sample_type

### If you need to change the metadata / add columns do it here so it's consistent across analysis files!
### Relevel factors here!
s <- read_qiime_mapping_file(mapping_file_fp) %>%
  mutate(isControl = grepl("emptywell|extractblank|dnafree|geneblock|mockdna", SampleID, ignore.case = T)) %>%
  rename(subject_id = SubjectID, sample_type = SampleType) %>%
  
  # One subject in control group has a Barrett's brushing sample
  mutate(sample_type = ifelse(
    (subject_id %in% "B2C007") & 
      (sample_type %in% "Barrett's brushing"),
    "Cardia brushing", sample_type)) %>%
  mutate(study_group = str_replace(
    study_group, "Control nonPPI", "Control non PPI")) %>%
  mutate(study_group = str_replace(
    study_group, "Control-non PPI", "Control non PPI")) %>%
  mutate(study_group = str_replace(
    study_group, "Control-PPI", "Control PPI")) %>%
  # Fix sample_types
  mutate(sample_type = str_replace(
    sample_type, "Blank [Ss]aliva [Kk]it", "Blank saliva kit")) %>%
  mutate(sample_type = str_replace(
    sample_type, "Blank swab", "Blank tip")) %>%
  mutate(sample_type = str_replace(
    sample_type, "Oral rinse( control)?", "Oral wash control")) %>%
  mutate(sample_type = str_replace(
    sample_type, "Barretts brushing", "Barrett's brushing")) %>%
  left_join(sample_types, by="sample_type") %>%
  mutate(control_label = if_else(
    is_control, "Control samples", "Experimental samples")) %>%
  mutate(sample_type = fct_relevel(sample_type, sample_type_order)) %>%
  # Rename esophagus brushing to squamous
  mutate(sample_type = fct_recode(
    sample_type,
    "Squamous brushing" = "Esophagus brushing")) %>%
  mutate(sample_type_collapse = fct_collapse(
    sample_type,
    "Barrett's/Cardia brushing" = c(
      "Barrett's brushing", "Cardia brushing"))) %>%
  mutate(sample_type_column = fct_recode(
    sample_type_collapse,
    Oral = "Oral wash",
    Squamous = "Squamous brushing",
    BarrettsCardia = "Barrett's/Cardia brushing")) %>%
  mutate(study_group_collapse = fct_collapse(
    study_group,
    "Barrett's" = c("BE LGD worse", "BE No dys indef"),
    "Healthy" = c("Control non PPI", "Control PPI"))) %>%
  mutate(on_ppi = study_group %in% c(
    "BE LGD worse", "BE No dys indef", "Control PPI")) %>%
  #mutate(final_library_conc_ng_ul = case_when(
  #  run %in% "Run7" ~ library_conc_ng_ul,
  #  run %in% "Run8" ~ final_conc_ng_ul,
  #  TRUE ~ final_library_conc_ng_ul)) %>%
  mutate(final_library_conc_ng_ul = pmax(final_library_conc_ng_ul, 0)) %>%
  
  left_join(clinical_data, by="subject_id")
```


```{r, warning = F}
### read otu count data
counts <- read_tsv(file = feature_table_fp, skip = 1) %>%
  column_to_rownames(var = "#OTU ID") %>%
  as.matrix()

### taxonomy assignment
ta <- read_tsv(file = taxo_assignment_fp) %>%
  mutate(Taxon = str_remove(Taxon, "(; [kpcofgs]__)+$")) %>%
  arrange(order(match(rownames(counts), `Feature ID`))) # make sure that OTU table and taxonomy assignment have the same rownames

### taxonomy assignment as a data frame
adf <- split_assignments(ta$Taxon)
rownames(adf) <- ta$`Feature ID`

#rm(ta)
```

```{r}
### get read counts after demultiplexing
#demux <- read_csv(file = demux_count_fp) %>%
#  setNames(c("SampleID", "demux_Read_Counts"))

### get read counts after denosing by DADA2 in QIIME2 pipeline
denoise <- colSums(counts) %>%
  enframe("SampleID", "denoise_Read_Counts")
  
### get read counts after removing contamination
is_mitochondrial <- grepl("mitochondria", adf$Family)
is_chloroplast <- grepl("Chloroplast", adf$Class)
is_unassigned <- grepl("Unassigned", adf$Kingdom)
#is_archaea <- grepl("Archaea", adf$Kingdom)
is_contam <- is_mitochondrial | is_chloroplast | is_unassigned 

counts <- counts[!is_contam, ]
adf <- adf[!is_contam, ]
ta <- ta[!is_contam,]

qc <- colSums(counts) %>%
  enframe("SampleID", "QC_Read_Counts") 

s <- s %>%
  #left_join(demux, by = "SampleID") %>%
  left_join(denoise, by = "SampleID") %>%
  left_join(qc, by = "SampleID") %>%
  mutate(above_min_reads = QC_Read_Counts > min_reads) %>%
  mutate(QC_read_call = factor(ifelse(above_min_reads, "above threshold", "below threshold"))) %>%
  mutate(Keep = !is.na(QC_Read_Counts) & QC_Read_Counts > min_reads) %>%
  mutate(Keep = ifelse(FINAL_WITHDRAW %in% "No (patient completely withdrawn from study)", FALSE, Keep))

rm(is_mitochondrial, is_chloroplast, is_unassigned)
rm(demux, denoise, qc)
```


```{r}
a <- simplify_assignments(adf, rank1="Phylum", rank2="Genus")
names(a) <- rownames(adf)
summed_cts <- rowsum(counts, a) 
summed_props <- sweep(summed_cts, 2, colSums(summed_cts), "/")

otu_props <- sweep(counts, 2, colSums(counts), "/")


a_f <- simplify_assignments(adf, rank1="Phylum", rank2="Family")
names(a_f) <- rownames(adf)
summed_cts_f <- rowsum(counts, a_f) 
summed_props_f <- sweep(summed_cts_f, 2, colSums(summed_cts_f), "/")

a_p <- simplify_assignments(adf, rank1="Kingdom", rank2="Phylum")
names(a_p) <- rownames(adf)
summed_cts_p <- rowsum(counts, a_p) 
summed_props_p <- sweep(summed_cts_p, 2, colSums(summed_cts_p), "/")
```


```{r}
library(mirix)



phenotypes <- ta %>%
  select(lineage="Taxon") %>%
  unique() %>%
  mutate(gram_stain = phenotype_susceptibility(lineage = lineage, phenotype="gram_stain", susceptibility = c(`Gram-positive` = "susceptible", `Gram-negative` = "resistant"), db = taxon_phenotypes)) %>%
  mutate(gram_stain = factor(gram_stain)) %>%
  mutate(gram_stain = fct_recode(gram_stain, `Gram-positive` = "susceptible", `Gram-negative` = "resistant")) %>%
  mutate(gram_stain = fct_explicit_na(gram_stain, na_level="Not annotated")) %>%
  
  mutate(obligate_anaerobe = phenotype_susceptibility(lineage = lineage, phenotype="aerobic_status", susceptibility = c(`obligate anaerobe` = "susceptible", `facultative anaerobe` = "resistant", `aerobe` = "resistant"), db = taxon_phenotypes)) %>%
  mutate(obligate_anaerobe = ifelse(obligate_anaerobe%in%"susceptible", "obligate_anaerobe", NA)) %>%
  
  mutate(facultative_anaerobe = phenotype_susceptibility(lineage = lineage, phenotype="aerobic_status", susceptibility = c(`obligate anaerobe` = "resistant", `facultative anaerobe` = "susceptible", `aerobe` = "resistant"), db = taxon_phenotypes)) %>%
  mutate(facultative_anaerobe = ifelse(facultative_anaerobe%in%"susceptible", "facultative_anaerobe", NA)) %>%
  
  mutate(aerobe = phenotype_susceptibility(lineage = lineage, phenotype="aerobic_status", susceptibility = c(`obligate anaerobe` = "resistant", `facultative anaerobe` = "resistant", `aerobe` = "susceptible"), db = taxon_phenotypes)) %>%
  mutate(aerobe = ifelse(aerobe%in%"susceptible", "aerobe", NA)) %>%
  
  mutate(aerobic_status = obligate_anaerobe) %>%
  mutate(aerobic_status = ifelse(!is.na(facultative_anaerobe), facultative_anaerobe, aerobic_status)) %>%
  mutate(aerobic_status = ifelse(!is.na(aerobe), aerobe, aerobic_status)) %>%
  mutate(aerobic_status = fct_explicit_na(aerobic_status, na_level="Not annotated")) %>%
  
  select(lineage, gram_stain, aerobic_status) %>%
  
  right_join(select(ta, Taxa="Feature ID", lineage="Taxon"), by="lineage") %>%
  select(-lineage)



props_with_phenotypes <- otu_props %>%
  as.data.frame() %>% 
  rownames_to_column("Taxa") %>% 
  pivot_longer(-Taxa, names_to="SampleID", values_to="proportion") %>%
  left_join(phenotypes, by="Taxa") %>%
  #filter(proportion<1) %>% arrange(-proportion) %>% slice(1:5) %>%
  pivot_longer(cols=-c("Taxa", "SampleID", "proportion"), names_to = "Phenotype_category", values_to = "Phenotype") %>%
  group_by(SampleID, Phenotype_category, Phenotype) %>%
  summarize(props = sum(proportion)) %>%
  ungroup() %>%
  mutate(Phenotype = gsub("_", " ", Phenotype))
  


```

```{r eval=F}
# sum bacterial abundances by phenotype
phenotypes <- read_delim("/Users/tanesc/Documents/DBs/bacteria_phenotypes/genera_0831.txt") %>%
  rename(Taxa = name)

props_with_phenotypes <- summed_props %>%
  as.data.frame() %>% 
  rownames_to_column("Taxa") %>% 
  pivot_longer(-Taxa, names_to="SampleID", values_to="props") %>%
  mutate(Taxa_old = Taxa) %>%
  mutate(Taxa = sub("p__.* [cofgs]__", "", Taxa_old)) %>%
  
  left_join(phenotypes, by="Taxa") %>%
  select(SampleID, props, aerobic_status:saccharolytic) %>%
  #filter(props > 0) %>%
  pivot_longer(cols=-c("SampleID", "props"), names_to = "Phenotype_category", values_to = "Phenotype") %>%
  mutate(Phenotype = factor(Phenotype)) %>%
  mutate(Phenotype = fct_explicit_na(Phenotype, na_level="Not annotated")) %>%
  group_by(SampleID, Phenotype_category, Phenotype) %>%
  summarize(props = sum(props)) %>%
  ungroup()
  
```


```{r alpha diversity}

richness <- rarefy(t(counts), richness_subsample_size) %>%
  enframe("SampleID", "Richness")
shannon <- diversity(t(counts)) %>%
  enframe("SampleID", "Shannon")
faith <- read_tsv(file = faith_fp) %>%
  setNames(c("SampleID", "Faith"))

### add alpha diversity measures
s <- s %>%
  left_join(richness, by = "SampleID") %>%
  left_join(shannon, by = "SampleID") %>%
  left_join(faith, by = "SampleID")

rm(richness, shannon, faith)
```  

```{r beta diversity}
wu <- read_qiime_distmat(wu_fp)
uu <- read_qiime_distmat(uu_fp)
```




# Quality control

## Total read pairs

Will remove 1 sample with >800,000 reads.

```{r}
s %>%
  filter(!is.na(QC_Read_Counts)) %>%
  ggplot(aes(x=QC_Read_Counts, fill=sample_type)) +
    geom_histogram(binwidth=1e4, boundary=0) +
    scale_y_continuous() +
    theme_clean()


s %>%
  filter(QC_Read_Counts > 800000) %>%
  select(SampleID, sample_type, QC_Read_Counts) %>%
  pander(split.table=Inf, digits=2)


s <- s %>%
  mutate(Keep = ifelse(QC_Read_Counts > 800000, F, Keep))

```


## Amount of potive control reads in samples

```{r fig.width=8}
pos_control_levels <- summed_props %>%
  as.data.frame() %>% 
  rownames_to_column("Taxa") %>% 
  pivot_longer(-Taxa, names_to="SampleID", values_to="props") %>%
  right_join(s, by="SampleID")  %>%
  filter(grepl("p__Euryarchaeota|p__Crenarchaeota", Taxa)) %>%
  group_by(SampleID, sample_type) %>%
  summarize(props = sum(props)) %>%
  ungroup() 

pos_control_levels %>%
  mutate(sample_type = fct_relabel(sample_type, function(x) sub(" ", "\n", x))) %>%
  ggplot(aes(x=sample_type, y=props)) +
    geom_boxplot(outlier.alpha = 0) +
    geom_quasirandom() +
    theme_clean()


sample_with_high_pos <- pos_control_levels %>%
  filter(!sample_type %in% c("DNA-free water", "Mock DNA")) %>%
  filter(props > 0.75) %>%
  select(SampleID, sample_type, pos_control_abundance=props) 

sample_with_high_pos %>%
  pander(split.table=Inf, digits=2)


s <- s %>%
  mutate(Keep = ifelse(SampleID %in% sample_with_high_pos$SampleID, F, Keep))
```




```{r}
## load in unassigner results
unassigner <- read_delim(file.path(data_dir, "qza_files/all_BetrNet_rep-seqs/dna-sequences_unassigned", "algorithm_output.tsv")) %>%
  filter(!is.na(probability_incompatible)) %>%
  filter(probability_incompatible < 0.5) %>%
  arrange(query_id, probability_incompatible) %>%
  
  group_by(query_id) %>%
  summarize(species_name = paste(species_name, collapse=", ")) %>%
  ungroup() 

a_with_unassigner <- a %>%
  merge(unassigner, by.y="query_id", by.x="row.names", all.x=T) %>%
  mutate(tax_assignment = ifelse(!is.na(species_name), paste("U", species_name), paste("Q", gsub("[pcofgs]__", "", x) ))) %>%
  select(Taxa = Row.names, tax_assignment)
```


```{r}
# Output files for collaborators

# Write out the updated mapping file
write.table(s, file=sub(".tsv", "_forCollaborators.tsv", mapping_file_fp), sep='\t', quote=F, row.names=F)


# Write out the list of excluded samples
write.table(pull(filter(s, !Keep & !isControl), SampleID), file=file.path(data_dir, "excluded_samples.tsv"), quote=F, row.names=F, col.names=F)

```

```{r eval=F}
## finding the saliva samples to send to collaborators
temp <- s %>% 
  filter(sample_type %in% c("Saliva", "Blank tip", "DNA-free water", "Empty well")) #%>% 
  #select(SampleID, BarcodeSequence, sample_type, subject_id, run, flow_cell_id, on_ppi, ANTIBIO, FINAL_PATH)

temp %>%
  write.table(file="betrnet_saliva.txt", sep='\t', row.names=F, quote=F)

runs <- unique(temp$run)

for (i in runs) {
  print(i)
  temp %>%
    filter(run == i) %>%
    droplevels() %>%
    write.table(file=paste0("betrnet_",i, ".txt"), sep='\t', row.names=F, quote=F)
}
```
